packages/jsii/lib/compiler.ts

```diff
@@ -178,7 +178,18 @@ export async function compileSources(entrypoint: string,
 
     if (rootModule) {
         await processModule(rootModule, []);
-        addTypeInfo(assm, types);
+        for (const type of types.sort((l, r) => l.fqn.localeCompare(r.fqn))) {
+            assm.types = assm.types || {};
+            if (type.namespace) {
+                const ns = `${type.assembly}.${type.namespace}`;
```

Comment:
Can this be implemented as a validation function like the ones below?

Reply:
@rix0rrr  came up with the same suggestion. Great minds think alike?